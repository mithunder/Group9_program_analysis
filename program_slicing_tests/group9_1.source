
module division_by_zero_program :

	read y;
	x := 0;
	z := 30;

	if (x = 0) -> x := 10
	[] (z > 0) -> z := -1
	fi;
	
	i := 0;
	sum := 0;
	do (i < 30) -> sum := sum+1
	od;

	a := 0;
	if (true) -> a := 30
	[] (true) -> a := 15
	fi;

	i := 0;

	do (i < 10) ->
	@programslicing="true"
		y := y / x; i := i+1
	od;

	write y*z;
	write sum*a
end

