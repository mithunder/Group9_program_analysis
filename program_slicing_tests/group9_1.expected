
module division_by_zero_program :

	read y;
	x := 0;
	z := 30;

	if (x = 0) -> x := 10
	[] (z > 0) -> skip
	fi;

	i := 0;

	do (i < 10) -> y := y / x; i := i+1
	od
end

